### DISCUSSION TITLE
support custom reader conditionals

### DISCUSSION THREAD

**Discussion by alidcast** [squint-cljs/cherry#84] [Ideas]
This will allow using Cherry in multiple environments and for cross-platform applications. It's the only feature I find missing coming from shadow-cljs (https://shadow-cljs.github.io/docs/UsersGuide.html#_conditional_reading).

Basic example of usage: 
```
#?(:web  "My computer!"
   :mobile "My phone!")
```

By the way, I got Cherry working with Expo's new file system Router (https://expo.github.io/router/docs) along with Nextjs. So new capabilities are truly opening up for leveraging best tools in JS ecosystem. Custom reader conditionals will allow us to all the way to writing cross-platform apps.

**Comment by alidcast** [comment]
@borkdude let me know if you'd prefer these issues in https://github.com/squint-cljs/compiler-common

**Comment by borkdude** [comment]
@aleedmanath Good to hear! 

The issue here is fine, but if/when we implement this, it will probably be the same for squint as well.
How does one solve the above issue in a pure JS environment?

**Comment by alidcast** [comment]
In JS it is done based on file names.  Example: `Component.web.js` (for web), `Component.ios.js` (for IOS), and `Component.js` (default) etc 

Swift recently introduced [Conditional Compilation Block](https://docs.swift.org/swift-book/ReferenceManual/Statements.html), which is similar to how it's done with reader conditionals in Clojure

**Comment by borkdude** [comment]
I guess for now the JS way would also work with cherry/squint?

**Comment by alidcast** [comment]
I'll have to try it, didn't consider it since Clojure/Script is finicky with file/namespace names

Earlier this year I spent some time seeing how Shadow-cljs implemented this, in case linking to code would be helpful for discussion. Though I understand if this is not a priority right now!

**Comment by borkdude** [comment]
I'll leave this issue open so we can collect more feedback from more users. I think the JS way would work with cherry / squint too.
